### ISSUE TITLE
test_post_training_quantization_mobilenetv1 UT fails in Paddle 2.4

### ISSUE THREAD

**Issue by leo0519** [PaddlePaddle/Paddle#48890]
### bug描述 Describe the Bug

- 版本、环境信息：
    1. PaddlePaddle版本：2.4.0
    2. CPU/GPU：A100
    3. Docker：paddlepaddle/paddle:2.4.0-gpu-cuda11.7-cudnn8.4-trt8.4
    4. Python版本号：3.7
    5. 显存信息

```bash
$ python tools/summary_env.py
****************************************
Paddle version: 2.4.0
Paddle With CUDA: True

OS: Ubuntu 18.04
Python version: 3.7.14

CUDA version: 11.7.64
Build cuda_11.7.r11.7/compiler.31294372_0
cuDNN version: None.None.None
Nvidia driver version: 525.57
****************************************
```

- 复现步骤

```bash
docker run -it --rm --gpus all paddlepaddle/paddle:2.4.0-gpu-cuda11.7-cudnn8.4-trt8.4 bash -c 'git clone https://github.com/PaddlePaddle/Paddle.git --branch release/2.4; python Paddle/paddle/fluid/contrib/slim/test/test_post_training_quantization_mobilenetv1.py'
```

-问题描述：请详细描述您的问题，同步贴出报错信息、日志/代码关键片段
There are two errors in the UT `test_post_training_quantization_mobilenetv1`

The first error log is as follow (first part). This error can be solved by this [PR](https://github.com/PaddlePaddle/Paddle/pull/46345), but it is not cherry-picked into Paddle 2.4.

After cherry-picking the PR, a new error will be reproduced. Refer to the second part of error log.

The first part of error log:
```bash
Cloning into 'Paddle'...
remote: Enumerating objects: 493681, done.
remote: Counting objects: 100% (347/347), done.
remote: Compressing objects: 100% (277/277), done.
remote: Total 493681 (delta 180), reused 121 (delta 70), pack-reused 493334
Receiving objects: 100% (493681/493681), 306.08 MiB | 20.33 MiB/s, done.
Resolving deltas: 100% (414920/414920), done.
Cache file /root/.cache/paddle/dataset/int8/download/calibration_test_data.tar.gz not found, downloading http://paddle-inference-dist.bj.bcebos.com/int8/calibration_test_data.tar.gz
Begin to download
item 3307/3307 [============================>.] - ETA: 0s - 4ms/item
Download finished
Data is downloaded at /root/.cache/paddle/dataset/int8/download/calibration_test_data.tar.gz
Cache file /root/.cache/paddle/dataset/int8/download/mobilenetv1_int8_model.tar.gz not found, downloading http://paddle-inference-dist.bj.bcebos.com/int8/mobilenetv1_int8_model.tar.gz
Begin to download
item 3894/3894 [============================>.] - ETA: 0s - 5ms/item
Download finished
Data is downloaded at /root/.cache/paddle/dataset/int8/download/mobilenetv1_int8_model.tar.gz
Start FP32 inference for MobileNet-V1 on 100 images ...
Paddle/python/paddle/fluid/contrib/slim/tests/test_post_training_quantization_mobilenetv1.py:48: DeprecationWarning: LANCZOS is deprecated and will be removed in Pillow 10 (2023-07-01). Use Resampling.LANCZOS instead.
  img = img.resize((resized_width, resized_height), Image.LANCZOS)
Start INT8 post training quantization for MobileNet-V1 on 100 images ...
EData is downloaded at /root/.cache/paddle/dataset/int8/download/calibration_test_data.tar.gz
Data is downloaded at /root/.cache/paddle/dataset/int8/download/mobilenetv1_int8_model.tar.gz
Start FP32 inference for MobileNet-V1 on 100 images ...
Start INT8 post training quantization for MobileNet-V1 on 100 images ...
EData is downloaded at /root/.cache/paddle/dataset/int8/download/calibration_test_data.tar.gz
Data is downloaded at /root/.cache/paddle/dataset/int8/download/mobilenetv1_int8_model.tar.gz
Start FP32 inference for MobileNet-V1 on 100 images ...
Start INT8 post training quantization for MobileNet-V1 on 100 images ...
EData is downloaded at /root/.cache/paddle/dataset/int8/download/calibration_test_data.tar.gz
Data is downloaded at /root/.cache/paddle/dataset/int8/download/mobilenetv1_int8_model.tar.gz
Start FP32 inference for MobileNet-V1 on 100 images ...
Start INT8 post training quantization for MobileNet-V1 on 100 images ...
EData is downloaded at /root/.cache/paddle/dataset/int8/download/calibration_test_data.tar.gz
Data is downloaded at /root/.cache/paddle/dataset/int8/download/mobilenetv1_int8_model.tar.gz
Start FP32 inference for MobileNet-V1 on 100 images ...
Start INT8 post training quantization for MobileNet-V1 on 100 images ...
E
======================================================================
ERROR: test_post_training_abs_max_mobilenetv1 (__main__.TestPostTrainingAbsMaxForMobilenetv1)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "Paddle/python/paddle/fluid/contrib/slim/tests/test_post_training_quantization_mobilenetv1.py", line 437, in test_post_training_abs_max_mobilenetv1
    is_optimize_model, diff_threshold)
  File "Paddle/python/paddle/fluid/contrib/slim/tests/test_post_training_quantization_mobilenetv1.py", line 311, in run_test
    onnx_format)
TypeError: generate_quantized_model() takes from 4 to 11 positional arguments but 12 were given

======================================================================
ERROR: test_post_training_onnx_format_mobilenetv1 (__main__.TestPostTrainingAvgONNXFormatForMobilenetv1)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "Paddle/python/paddle/fluid/contrib/slim/tests/test_post_training_quantization_mobilenetv1.py", line 472, in test_post_training_onnx_format_mobilenetv1
    batch_nums=batch_nums)
  File "Paddle/python/paddle/fluid/contrib/slim/tests/test_post_training_quantization_mobilenetv1.py", line 311, in run_test
    onnx_format)
TypeError: generate_quantized_model() takes from 4 to 11 positional arguments but 12 were given

======================================================================
ERROR: test_post_training_kl_mobilenetv1 (__main__.TestPostTrainingKLForMobilenetv1)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "Paddle/python/paddle/fluid/contrib/slim/tests/test_post_training_quantization_mobilenetv1.py", line 356, in test_post_training_kl_mobilenetv1
    is_optimize_model, diff_threshold)
  File "Paddle/python/paddle/fluid/contrib/slim/tests/test_post_training_quantization_mobilenetv1.py", line 311, in run_test
    onnx_format)
TypeError: generate_quantized_model() takes from 4 to 11 positional arguments but 12 were given

======================================================================
ERROR: test_post_training_avg_mobilenetv1 (__main__.TestPostTrainingavgForMobilenetv1)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "Paddle/python/paddle/fluid/contrib/slim/tests/test_post_training_quantization_mobilenetv1.py", line 380, in test_post_training_avg_mobilenetv1
    is_optimize_model, diff_threshold)
  File "Paddle/python/paddle/fluid/contrib/slim/tests/test_post_training_quantization_mobilenetv1.py", line 311, in run_test
    onnx_format)
TypeError: generate_quantized_model() takes from 4 to 11 positional arguments but 12 were given

======================================================================
ERROR: test_post_training_hist_mobilenetv1 (__main__.TestPostTraininghistForMobilenetv1)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "Paddle/python/paddle/fluid/contrib/slim/tests/test_post_training_quantization_mobilenetv1.py", line 413, in test_post_training_hist_mobilenetv1
    batch_nums=batch_nums)
  File "Paddle/python/paddle/fluid/contrib/slim/tests/test_post_training_quantization_mobilenetv1.py", line 311, in run_test
    onnx_format)
TypeError: generate_quantized_model() takes from 4 to 11 positional arguments but 12 were given

----------------------------------------------------------------------
Ran 5 tests in 68.257s

FAILED (errors=5)
```

The second part of error log:
```bash
======================================================================
ERROR: test_post_training_onnx_format_mobilenetv1 (__main__.TestPostTrainingAvgONNXFormatForMobilenetv1)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test_post_training_quantization_mobilenetv1.py", line 458, in test_post_training_onnx_format_mobilenetv1
    self.run_test(model,
  File "test_post_training_quantization_mobilenetv1.py", line 303, in run_test
    self.generate_quantized_model(os.path.join(model_cache_folder, "model"),
  File "test_post_training_quantization_mobilenetv1.py", line 274, in generate_quantized_model
    ptq.quantize()
  File "/usr/local/lib/python3.8/dist-packages/paddle/fluid/contrib/slim/quantization/post_training_quantization.py", line 441, in quantize
    self._update_program()
  File "/usr/local/lib/python3.8/dist-packages/paddle/fluid/contrib/slim/quantization/post_training_quantization.py", line 1111, in _update_program
    self._program = graph.to_program()
  File "/usr/local/lib/python3.8/dist-packages/paddle/fluid/framework.py", line 5186, in to_program
    convert_pass.apply(self.graph)
ValueError: (InvalidArgument) Generated graph shouldn't contain cycle.
  [Hint: Expected HasCircleInternal<DescOrderComparator>(adj_list, nullptr) == false, but received HasCircleInternal<DescOrderComparator>(adj_list, nullptr):1 != false:0.] (at /opt/paddle/paddle/paddle/fluid/framework/ir/graph_helper.cc:437)
----------------------------------------------------------------------
```

**Comment by leo0519** [comment]
Hi @yghstill. Is there any update about this issue?

**Comment by yghstill** [comment]
Fixd in #49257

**Comment by leo0519** [comment]
Verified that it is fixed in Paddle release 2.4.
